### PR TITLE
israel.c/ api + service & tests

### PR DESCRIPTION
**Pull request by IsraelCohen54**
Hello!
I am trying to make it shorted for you:
1. Only the last 2 functions inside controller & PetService & PetServiceTest are relevant.
2. Only the last func inside PetDao & PetDaoTest is relevant (getOwnerPets).

### CODE REVIEW

File: src/test/kotlin/com/hibob/academy/service/PetServiceTest.kt [hunk 1]
```diff
@@ -66,4 +71,40 @@ class PetServiceTest {
...
+
+        whenever(petDao.getOwnerPets(eq(ownerId))).thenReturn(pets)
+
+        val result = petService.getOwnerPets(ownerId)
+
+        assertEquals(pets, result)
+    }
+
+    @Test
+    fun `test countPetsByType returns correct pet counts`() {
```
Comment (IsraelCohen54) [inline src/test/kotlin/com/hibob/academy/service/PetServiceTest.kt]: I'm using another func from petDao (getPetsByType) to adhere to DRY principle, so I didn't create a new func inside petDao for counting.
It seemed close enough for me.

File: src/test/kotlin/com/hibob/academy/service/PetServiceTest.kt [hunk 2]
```diff
@@ -1,25 +1,30 @@
...
 import org.mockito.kotlin.eq
 import org.mockito.kotlin.whenever
+import java.time.LocalDate
+import kotlin.random.Random
 
 class PetServiceTest {
 
     private val petDao: PetDao = mock(PetDao::class.java)
     private val petService = PetService(petDao)
+    private val companyId = Random.nextLong()
```
Comment (IsraelCohen54) [inline src/test/kotlin/com/hibob/academy/service/PetServiceTest.kt:23]: Better placed here and defined once, needed for the usage of petDao function

File: src/main/kotlin/com/hibob/academy/service/PetService.kt [hunk 3]
```diff
@@ -18,4 +20,17 @@ class PetService(private val petDao: PetDao) {
         val owner = petDao.getPetOwner(petId)
         return owner ?: throw IllegalStateException("No owner with ID $petId")
     }
+
+    fun getOwnerPets(ownerId: Long): List<Pet> {
+        return petDao.getOwnerPets(ownerId)
+    }
+
+    fun countPetsByType(companyId: Long): Map<PetType, Int> {
```
Comment (IsraelCohen54) [inline src/main/kotlin/com/hibob/academy/service/PetService.kt:28]: I'm using another func from petDao (getPetsByType) to adhere to DRY principle, so I didn't create a new func inside petDao for counting.
It seemed close enough for me.

File: src/main/kotlin/com/hibob/academy/service/PetService.kt [hunk 4]
```diff
@@ -18,4 +20,17 @@ class PetService(private val petDao: PetDao) {
...
+        val petTypes = PetType.entries.toTypedArray()
+
+        return petTypes.associateWith { type ->
+            val pets = petDao.getPetsByType(type, companyId)
```
Comment (IsraelCohen54) [inline src/main/kotlin/com/hibob/academy/service/PetService.kt:32]: I'm using another func from petDao (getPetsByType) to adhere to DRY principle, so I didn't create a new func inside petDao for counting.
It seemed close enough for me.

File: src/main/kotlin/com/hibob/academy/service/PetService.kt (hunk 3 above)
Comment (IsraelCohen54) [inline src/main/kotlin/com/hibob/academy/service/PetService.kt:28]: I'm using another func from petDao (getPetsByType) to adhere to DRY principle, so I didn't create a new func inside petDao for counting.
It seemed close enough for me.

File: src/main/kotlin/com/hibob/academy/dao/PetDao.kt [hunk 5]
```diff
@@ -85,4 +85,12 @@ class PetDao @Inject constructor(private val sql: DSLContext) {
                 )
             }
     }
+
+    fun getOwnerPets(ownerId: Long, companyId: Long): List<Pet> {
```
Comment (mayhuberman) [inline src/main/kotlin/com/hibob/academy/dao/PetDao.kt]: better getPetsByOwnerId

File: src/main/kotlin/com/hibob/academy/resource/ex10-controler.kt [hunk 6]
```diff
@@ -1,18 +1,20 @@
...
 import jakarta.ws.rs.core.Response.Status
-
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RequestParam
 
 @Path("/api")
 @Produces("application/json")
 @Consumes("application/json")
 class DaoController @Inject constructor(private val petService: PetService) {
```
Comment (mayhuberman) [inline src/main/kotlin/com/hibob/academy/resource/ex10-controler.kt]: should be PetsController

File: src/main/kotlin/com/hibob/academy/resource/ex10-controler.kt [hunk 7]
```diff
@@ -1,18 +1,20 @@
...
 import jakarta.inject.Inject
 import jakarta.ws.rs.*
 import jakarta.ws.rs.core.Response
 import jakarta.ws.rs.core.Response.Status
-
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RequestParam
 
 @Path("/api")
```
Comment (mayhuberman) [inline src/main/kotlin/com/hibob/academy/resource/ex10-controler.kt]: should be @Path("/api/pets")

File: src/main/kotlin/com/hibob/academy/resource/ex10-controler.kt [hunk 8]
```diff
@@ -74,4 +76,18 @@ class DaoController @Inject constructor(private val petService: PetService) {
         val owner = petService.getOwnerByPetId(petId, companyId)
            return Response.ok(owner).build()
     }
+
+    @GET
+    @Path("/{companyId}/{ownerId}")
```
Comment (mayhuberman) [inline src/main/kotlin/com/hibob/academy/resource/ex10-controler.kt]: how would the client know what is the companyId and what is the ownerId
change to     @Path("/company/{companyId}/owner/{ownerId}")

File: src/main/kotlin/com/hibob/academy/resource/ex10-controler.kt [hunk 9]
```diff
@@ -74,4 +76,18 @@ class DaoController @Inject constructor(private val petService: PetService) {
...
+    fun getOwnerPets(@PathParam("ownerId") ownerId: Long, @PathParam("companyId") companyId: Long): Response {
+        val ownerPets = petService.getOwnerPets(ownerId, companyId)
+        return Response.ok(ownerPets).build()
+    }
+
+    @GetMapping("/{companyId}/countByType")
```
Comment (mayhuberman) [inline src/main/kotlin/com/hibob/academy/resource/ex10-controler.kt]: better be consistent and use @GET and @Path annotations
and the path should be "/count-by-type/company/{companyId}"

File: src/main/kotlin/com/hibob/academy/resource/ex10-controler.kt [hunk 10]
```diff
@@ -74,4 +76,18 @@ class DaoController @Inject constructor(private val petService: PetService) {
...
+    fun countPetsByType(@RequestParam companyId: Long): ResponseEntity<Map<PetType, Int>> {
+        return ResponseEntity.ok(petService.countPetsByType(companyId))
```
Comment (mayhuberman) [inline src/main/kotlin/com/hibob/academy/resource/ex10-controler.kt]: again be consistent and retrun Response

File: src/main/kotlin/com/hibob/academy/service/PetService.kt [hunk 11]
```diff
@@ -18,4 +20,17 @@ class PetService(private val petDao: PetDao) {
         return petDao.getPetOwner(petId, companyId)
             ?: throw IllegalStateException("No owner with ID $petId")
     }
+
+    fun getOwnerPets(ownerId: Long, companyId: Long): List<Pet> {
```
Comment (mayhuberman) [inline src/main/kotlin/com/hibob/academy/service/PetService.kt]: better getPetsByOwnerId

File: src/main/kotlin/com/hibob/academy/service/PetService.kt (hunk 3 above)
Comment (mayhuberman) [reply src/main/kotlin/com/hibob/academy/service/PetService.kt:28]: you can do it by one method

File: src/main/kotlin/com/hibob/academy/resource/ex10-controler.kt [hunk 12]
```diff
@@ -74,4 +71,17 @@ class DaoController @Inject constructor(private val petService: PetService) {
         val owner = petService.getOwnerByPetId(petId, companyId)
            return Response.ok(owner).build()
     }
+
+    @GET
+    @Path("/company/{companyId}/owner/{ownerId}")
+    fun getOwnerPets(@PathParam("ownerId") ownerId: Long, @PathParam("companyId") companyId: Long): Response {
```
Comment (mayhuberman) [inline src/main/kotlin/com/hibob/academy/resource/ex10-controler.kt]: rename to getPetsByOwnerId

File: src/main/kotlin/com/hibob/academy/resource/ex10-controler.kt [hunk 13]
```diff
@@ -74,4 +71,17 @@ class DaoController @Inject constructor(private val petService: PetService) {
...
+        val ownerPets = petService.getPetsByOwnerId(ownerId, companyId)
```
Comment (mayhuberman) [inline src/main/kotlin/com/hibob/academy/resource/ex10-controler.kt]: rename to pets, you returning the pets not the owners

**Review by mayhuberman — APPROVED** [review]

**Pull request merged**